### DISCUSSION TITLE
Return the lock codes in lock attributes?

### DISCUSSION THREAD

**Discussion by irvintim** [jason0x43/hacs-hubitat#256] [Ideas]
Any chance I can convince you to remove the code that deletes the lock codes from the data returned to the lock attributes, or at least make it a conifgurable option to remove this or not. Specifically, it is this code in `locks.py` that I am referring to:
```python
if codes:
    for id in codes:
        del codes[id]["code"]
```

I need the codes in Home Assistant as I am managing the lock codes in HA and need to determine if a code is used before allowing it to be used again.  My workaround is to store the code in the Name field, which isn't great.  I understand the security concern here, so I'm suggesting this could be configurable.

**Comment by jason0x43** [comment]
An option not to hide the codes, or maybe a service call to retrieve the codes, seems reasonable.

**Reply by irvintim** [reply]
Do you want me to create an "Enhancement" Issue request?  
If I can get the time in the next day or 2, I can try to get you a "Pull Request" if that is helpful. Do you have a preference for the option to not hide vs. service call?
